pkg/flow/flow_mem_driver.go

```diff
@@ -823,11 +823,15 @@ func (fc *FlowCollector) updateRecord(record interface{}) error {
 							}
 						}
 
-						// find any links to the peer router and delete
+						// find any links to the peer router and delete the one that is for current
 						for _, eLink := range fc.Links {
 							if eLink.Direction != nil && *eLink.Direction == peerDirection {
 								if routerId == eLink.Parent {
-									fc.deleteRecord(eLink)
+									parts := strings.Split(*eLink.Name, "-")
+									if parts[len(parts)-1]+":0" == current.Parent {
```

Comment:
I do not believe this pattern holds for podman sites. The router will choose a random vanflow id prefix for itself if its configured name isn't pod name-like, so I don't think that name suffix matching logic would work.

I'd suggest the method Lynne took above:
Hold on to something like `currentRouterName := normalizeRouterName(*fc.Routers[current.Parent].Name)` then just compare that to eLink.Name. (also maybe nil-check the pointer dereferences - especially if connections are coming and going we may not have parents and/or names)